firmware/library/L3_HAL/apds_9960.hpp

```diff
@@ -1,20 +1,444 @@
+  virtual bool DisableGesture() = 0;
+  virtual uint8_t ReadGestureMode() = 0;
+  virtual bool CheckIfGestureOccured() = 0;
+  virtual uint8_t GetGestureFIFOLvl() = 0;
+  virtual void ReadGestureFIFO(uint8_t * data, uint8_t fifolvl) = 0;
+  virtual void ProcessGestureData(uint8_t level) = 0;
+  virtual char DecodeGestureData() = 0;
+  virtual char GetGesture() = 0;
+};
+
+class Apds9960 : public Apds9960Interface
+{
+ public:
+  // Apds-9960 Registers
+  static constexpr uint8_t
+    kENABLE = 0x80,
+    kATIME = 0x81,
+    kWTIME = 0x83,
+    kAILTL = 0x84,
+    kAILTH = 0x85,
+    kAIHTL = 0x86,
+    kAIHTH = 0x87,
+    kPILT = 0x89,
+    kPIHT = 0x8B,
+    kPERS = 0x8C,
+    kCONFIG1 = 0x8D,
+    kPPULSE = 0x8E,
+    kCONTROL = 0x8F,
+    kCONFIG2 = 0x90,
+    kID = 0x92,
+    kSTATUS = 0x93,
+    kCDATAL = 0x94,
+    kCDATAH = 0x95,
+    kRDATAL = 0x96,
+    kRDATAH = 0x97,
+    kGDATAL = 0x98,
+    kGDATAH = 0x99,
+    kBDATAL = 0x9A,
+    kBDATAH = 0x9B,
+    kPDATA =  0x9C,
+    kPoffsetUr = 0x9D,
+    kPoffsetDl = 0x9E,
+    kCONFIG3 = 0x9F,
+    kGPENTH = 0xA0,
+    kGEXTH =  0xA1,
+    kGCONF1 = 0xA2,
+    kGCONF2 = 0xA3,
+    kGoffsetU = 0xA4,
+    kGoffsetD = 0xA5,
+    kGoffsetL = 0xA7,
+    kGoffsetR = 0xA9,
+    kGPULSE = 0xA6,
+    kGCONF3 = 0xAA,
+    kGCONF4 = 0xAB,
+    kGFLVL =  0xAE,
+    kGSTATUS = 0xAF,
+    kIFORCE = 0xE4,
+    kPICLEAR = 0xE5,
+    kCICLEAR = 0xE6,
+    kAICLEAR = 0xE7,
+    kGFifoU = 0xFC,
+    kGFifoD = 0xFD,
+    kGFifoL = 0xFE,
+    kGFifoR = 0xFF;
+
+  enum class Direction : uint8_t
+  {
+    kNone = 0,
+    kUp,
+    kDown,
+    kLeft,
+    kRight,
+    kFar,
+    kNear
+  };
+  enum class Gesture : char
+  {
+    kSwipeUP = 'U',
+    kSwipeDOWN = 'D',
+    kSwipeLEFT = 'L',
+    kSwipeRIGHT = 'R',
+    kNEAR = 'N',
+    kFAR = 'F',
+    kERROR = 'E'
+  };
+
+  static constexpr uint8_t
+  kNoMode = 0x00,
+  kAtime = 0xDB, kWtime = 0xF6,
+  kAiltl = 0xFF, kAilth = 0xFF,
+  kAihtl = 0x00, kAihth = 0x00,
+  kPilt = 0x00, kPiht = 0x32,
+  kPers = 0x11, kConfig1 = 0x60,
+  kPpulse = 0x89,
+  kControl = 0x05,
+  kConfig2 = 0x01,
+  kPoffsetUpRight = 0x00,
+  kPoffsetDownLeft = 0x00,
+  kConfig3 = 0x00, kGPEnTh = 0x28,
+  kGExTh = 0x64, kGconfig1 = 0x40,
+  kGconfig2 = 0x66,
+  kGoffsetUpDown = 0x00,
+  kGoffsetLeftRight = 0x20,
+  kGPulse = 0xC5,
+  kGconfig3 = 0x00, kGconfig4 = 0x02;
+
+  void WriteReg(uint8_t reg, uint8_t data) override
+  {
+    uint8_t data_stream[2] = {reg, data};
+    i2c.Write(kApds9960Address, data_stream,
+        sizeof(data_stream));
+  }
+  uint8_t ReadReg(uint8_t reg) override
+  {
+    uint8_t read_data = 0;
+    i2c.WriteThenRead(kApds9960Address, &reg, 1, &read_data, 1);
+    return read_data;
+  }
+  void Initialize() override
+  {
+    I2cInterface::Status status;
+    i2c.Initialize();
+    status = i2c.Write(kApds9960Address, nullptr, 0);
+    if (status == I2cInterface::Status::kSuccess)
+    {
+      uint8_t initialization_sequence[56] =
```

Comment:
This should be a `static constexpr` array. This will keep the array from being loaded during runtime. It should be set in stone at the time of compilation.

Reply:
Might be good to take this out of this function and put it within the class scope.